x-pack/elastic-agent/pkg/agent/operation/operator.go

```diff
@@ -155,13 +157,20 @@ func (o *Operator) Close() error {
 	o.monitor.Close()
 	o.statusReporter.Unregister()
 
-	return o.HandleConfig(configrequest.New("", time.Now(), nil))
+	return o.HandleConfig(context.Background(), configrequest.New("", time.Now(), nil))
 }
 
 // HandleConfig handles configuration for a pipeline and performs actions to achieve this configuration.
-func (o *Operator) HandleConfig(cfg configrequest.Request) (err error) {
+func (o *Operator) HandleConfig(ctx context.Context, cfg configrequest.Request) (err error) {
+	// TODO: double check `route` as name
+	span, ctx := apm.StartSpan(ctx, "route", "app.internal")
 	defer func() {
-		err = filterContextCancelled(err)
+		if err != nil {
+			apm.CaptureError(ctx, err).Send()
+			// TODO: do we want to report `context.Canceled`?
```

Comment:
I assume that for tracing purposes this would be helpful, but let me know if you'd rather have it filtered out.

Reply:
If the errors are being wrapped then it would, but I'm not familiar enough with the codebase to know how the errors are being handled. The noise was also my main concern.

Reply:
Cool, I'll remove it for now